src/core/Val.ml

```diff
@@ -1199,12 +1199,8 @@ struct
           let face =
             Face.map @@ fun ri r'i absi ->
             let phi = I.equate ri r'i in
-            let phi0 = I.cmp (I.equate (`Atom x) `Dim0) phi in
             let yi, el = Abs.unleash absi in
-
-            (* If it helps, this right here is the source of the error that appears in omega1s1. - JMS *)
-            (* Format.eprintf "Here's a bad argument to rigid_vproj: %a@.@." pp_value el; *)
-            Abs.bind yi @@ rigid_vproj x ~ty0:(Value.act phi0 ty0) ~ty1:(Value.act phi ty1) ~equiv:(Value.act phi0 equiv) ~el
+            Abs.bind yi @@ vproj phi (I.act phi @@ `Atom x) ~ty0:(fun phi -> Value.act phi ty0) ~ty1:ty1 ~equiv:(fun phi -> Value.act phi equiv) ~el
```

Comment:
@favonia This change makes the `omega1s1` example go through, but I don't know if it is correct. Can you have a look?

Reply:
So it is basically the same issue. Thanks!

Reply:
How do you feel about this new continuation-based API?

Reply:
I think it's alright, but I don't yet have the right intuitions for it. I think generally, diagonals in the current implementation are very fucked up.